uktrade/data-hub-frontend
# Use resource for related company count

## Description of change

As we now have a dedicated API endpoint to retrieve the count of related companies, we can retrieve this information using the Resource component.

This is part of a larger piece of work to migrate the `CompanyLayout` component (and all pages using it) over to React Router.

## Test instructions
Go to a company page and view any tab. The DNB company count should display correctly.

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
